src/AspNetCoreModuleV2/CommonLib/GlobalVersionUtility.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "stdafx.h"
+#include <experimental/filesystem>
+
+// throws runtime error if no request handler versions are installed.
+std::wstring
+GlobalVersionUtility::GetGlobalRequestHandlerPath(PCWSTR pwzAspNetCoreFolderPath, PCWSTR pwzHandlerVersion, PCWSTR pwzHandlerName)
+{
+    std::wstring handlerVersion(pwzHandlerVersion);
```

Comment:
why not use folderVersion here? in line 15 you can check if folderVersion is empty and reset it and thus you don't need line 21 to copy the string.